Source/Models/APIClient.swift

```diff
@@ -2,19 +2,24 @@ import Foundation
 import Argo
 import Result
 
-public struct APIClient {
+public struct APIClient: Client {
   private let requestPerformer: RequestPerformer
 
+  public var queue: dispatch_queue_t? = dispatch_get_main_queue()
+
   public init(requestPerformer: RequestPerformer = NetworkRequestPerformer()) {
     self.requestPerformer = requestPerformer
   }
-}
 
-extension APIClient: Client {
   public func performRequest<T: Request>(request: T, completionHandler: Result<T.ResponseObject, SwishError> -> Void) -> NSURLSessionDataTask {
-    return requestPerformer.performRequest(request.build()) { result in
+    return requestPerformer.performRequest(request.build()) { [queue] result in
```

Comment:
I see, you're capturing here. I still think that leaving it mutable makes the API more complicated and I'd like to avoid that. If you need multiple queues, you can create multiple clients.